ember-scoped-css/src/build/scoped-css-unplugin.js

```diff
@@ -1,39 +1,122 @@
+import { readFileSync } from 'node:fs';
 import path from 'node:path';
+import { cwd } from 'node:process';
 
 import { createUnplugin } from 'unplugin';
 
-import { decodeScopedCSSRequest, isScopedCSSRequest } from '../lib/request.js';
+import { rewriteCss } from '../lib/css/rewrite.js';
+import { request } from '../lib/request.js';
+
 /**
- * The plugin that handles CSS requests from other transform (e.g.: babel)
+ * The plugin that handles CSS requests for `<style>` elements and transforms
+ * for existing files
  *
- * This plugin takes no options because the CSS transform has already happened by the time
- * we handle it here.
+ * vite: CSS files are resolved by vite. We use their resolver to also get
+ *       HMR. That is, for all non-physical CSS files, we extend vite by our
+ *       resolver and also can enrich metadata to it (for better debugging)
  */
-export default createUnplugin(() => {
-  return {
-    name: 'ember-scoped-css-unplugin',
-
-    resolveId(id, importer) {
-      if (isScopedCSSRequest(id)) {
-        let parsed = decodeScopedCSSRequest(id);
-
-        return {
-          id: path.resolve(path.dirname(importer), parsed.postfix + '.css'),
-          meta: {
-            'scoped-css': {
-              css: parsed.css,
+export default createUnplugin((options = {}) => {
+  const CWD = process.cwd();
+
+  return [
+    {
+      name: 'ember-scoped-css:colocated',
+      resolveId(id, importer) {
+        // handles: some-file.css?scoped=[postfix]
+        // this is only run in rollup, vite handles it differently
+        if (request.is.colocated(id)) {
+          const parsed = request.colocated.decode(id);
+
+          // This should end up pointing at the actual file on disk
+          const filePath = path.resolve(
+            path.dirname(importer),
+            path.basename(parsed.fileName),
+          );
+          const relativeFilePath = path.relative(CWD, filePath);
+          
+          let code = readFileSync(filePath, 'utf-8');
+          let css = rewriteCss(
+            code,
+            parsed.postfix,
+            relativeFilePath,
+            options.layerName,
+          );
```

Comment:
you wanna leave that in `transform()` - as this is what's happening. More of a semantic thing, but nevertheless